src/app/characters/character-list/character-list.component.spec.ts

```diff
@@ -99,16 +108,27 @@ describe(CharacterListComponent.name, () => {
   }));
 
   it('should select a character', async () => {
-    component.characters$ = of(fakeCharacters);
+    component.characters$ = of(fakeMarvelResponseData.results);
     fixture.detectChanges();
     const cardDisplay = fixture.debugElement.query(By.css('app-character-card'));
-    cardDisplay.triggerEventHandler('selectedChange', fakeCharacters[0]);
-    expect(component.selectedCharacter).toEqual(fakeCharacters[0]);
+    cardDisplay.triggerEventHandler('selectedChange', fakeMarvelResponseData.results[0]);
+    expect(component.selectedCharacter).toEqual(fakeMarvelResponseData.results[0]);
   });
 
   it('should display progress bar', () => {
     component.showProgress = true;
     fixture.detectChanges();
     expect(fixture.nativeElement.querySelector('mat-progress-bar')).not.toBeNull();
   });
+
+  it('should display warning message', fakeAsync(() => {
+    characterServiceSpy.getCharacters.and.returnValue(of({
+      ...fakeMarvelResponseData,
+      total: environment.settings.charactersLimit + 1
+    }));
+    search();
+    tick(300);
+    fixture.detectChanges();
+    expect(snackbarSpy.open.calls.first().args[0]).toContain('There are more results');
```

Comment:
Since the `open` method of the snackbar does not use any component related properties (it contains only hardcoded values), there is no point of testing its arguments (this is probably a better case for e2e testing). So in this case it would suffice just to test whether the open method is called.

Reply:
Done.